Celeste.Mod.mm/Mod/Everest/Everest.Loader.cs

```diff
@@ -484,12 +484,18 @@ public static class Loader {
                 Assembly asm = null;
                 if (!string.IsNullOrEmpty(meta.PathArchive)) {
                     bool returnEarly = false;
+                    string dllDirectory = Path.GetDirectoryName(meta.DLL) ?? "";
                     using (ZipFile zip = new ZipFile(meta.PathArchive)) {
                         foreach (ZipEntry entry in zip.Entries) {
                             string entryName = entry.FileName.Replace('\\', '/');
-                            if (entryName == meta.DLL) {
-                                using (MemoryStream stream = entry.ExtractStream())
-                                    asm = Relinker.GetRelinkedAssembly(meta, Path.GetFileNameWithoutExtension(meta.DLL), stream);
+
+                            if (Path.GetExtension(entryName) == ".dll") {
+                                if (entryName == meta.DLL)
+                                    using (MemoryStream stream = entry.ExtractStream())
+                                        asm = Relinker.GetRelinkedAssembly(meta, Path.GetFileNameWithoutExtension(meta.DLL), stream);
+                                else if (Path.GetDirectoryName(entryName) == dllDirectory) // immediately load any other libraries to avoid reading zip files over and over again later
+                                    using (MemoryStream stream = entry.ExtractStream())
+                                        Relinker.GetRelinkedAssembly(meta, Path.GetFileNameWithoutExtension(entryName), stream);
```

Comment:
I'm a bit paranoid about this conflicting with mod load order, multiple copies of the same dependency now being differently preferred, and "accidental" dependencies being forcibly loaded. Yet on the other hand, the existing code is horrible, thus if this change doesn't break any mods, I'm willing to accept it.